### PR TITLE
Keep headers at 4-level max, do minor structural corrections

### PR DESCRIPTION
**Pull request by spetrosi**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1962374

### CODE REVIEW

**Comment by spetrosi** [comment]
Hi @nhosoi, in this PR I restructure the readme a little to get rid of lvl 5+ headers.

Talking about md2html.sh, I noticed that pandoc produces weird `+` signed around inline code (what is marked with backtiks in MD). Then asciidoc keeps them after the conversion to HTML. This can be fixed easily by substituting asciidoc with asciidoctor - really only substitute the `s/asciidoc/asciidoctor/` in the script because asciidoctor uses all the same options that asciidoc does. Also, the HTML converted by asciidoctor is pretier than that converted by asciidoc. Can we make this change as part of your PR too https://src.fedoraproject.org/rpms/linux-system-roles/pull-request/11? See this PR: https://github.com/linux-system-roles/auto-maintenance/pull/71.

**Comment by spetrosi** [comment]
Pavel pointed me to the fact that asciidoctor is not available in RHE l7 and 8 buildroots, so we must use asciidoc as always.

File: README.md [hunk 1]
```diff
@@ -6,16 +6,15 @@
...
-    * [Brief overview](#brief-overview)
-    * [Variables](#variables)
-      * [Logging_inputs options](#logging_inputs-options)
-      * [Logging_outputs options](#logging_outputs-options)
-      * [Logging_flows options](#logging_flows-options)
-      * [Security options](#security-options)
-      * [Server performance optimization options](#server-performance-optimization-options)
-      * [Other options](#other-options)
-    * [Update and Delete](#update-and-delete)
+  * [Brief overview](#brief-overview)
```
Comment (nhosoi) [inline README.md:9]: You replaced `Brief overview` with `Logging Configuration Overview`?
https://github.com/linux-system-roles/logging/pull/228/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L61-R58

File: README.md [hunk 2]
```diff
@@ -56,9 +55,7 @@ This role is supported on RHEL/CentOS-7, RHEL/CentOS-8 and Fedora distributions.
 - [`Rsyslog`](https://www.rsyslog.com/) - The logging role default log provider used for log processing.
 - [`Elasticsearch`](https://www.elastic.co/) - Elasticsearch is a distributed, search and analytic engine for all types of data. One of the supported outputs in the logging role.
 
-## Logging Configuration
```
Comment (nhosoi) [inline README.md:59]: There is a link pointing this old header.
` 36 To satisfy such requirements, logging role introduced 3 primary variables `logging_inputs`, `l    ogging_outputs`, and `logging_flows`. The inputs are represented in the list of `logging_input    s` dictionary, the outputs are in the list of `logging_outputs` dictionary, and the relationsh    ip between them are defined as a list of `logging_flows` dictionary. The details are described     in [Logging Configuration](#logging-configuration).
`

**Comment by nhosoi** [comment]
Note: the restructuring to get rid of lvl 5+ headers is successful. I checked the results from both pandoc / kramdown as well as on AH. All look good. Thanks, @spetrosi!

**Review by nhosoi — APPROVED** [review]
lgtm

**Pull request merged**